File: app/src/main/java/configgen/value/ValueEnv.java
package configgen.value;

import configgen.data.CfgData;
import configgen.data.HeadRow;
import configgen.data.JsonTableFiles;
import configgen.i18n.LangTextFinder;
import configgen.schema.CfgSchema;

/// 值解析所需的全局环境（原直接持有ctx.Context，为解value→ctx依赖环改为参数对象）。
/// 由Context.makeValue在锁内构建快照，保证与解析时刻的数据一致。
public record ValueEnv(CfgSchema fullSchema,
                       CfgData cfgData,
                       HeadRow headRow,
                       LangTextFinder nullableLangTextFinder,
                       JsonTableFiles jsonTableFiles) {
}

File: packages/value/src/ValueEnv.ts
/**
 * ValueEnv — TypeScript port of Java `configgen.value.ValueEnv`.
 *
 * A parameter object holding all global state needed for value parsing.
 * Java original is a `record` (immutable); TS uses a class with readonly fields.
 *
 * Java source: configgen.value.ValueEnv.java (16 lines)
 */

import type { CfgSchema } from '@cfgforge/schema';
import type { CfgData } from '@cfgforge/data';
import type { HeadRow } from '@cfgforge/data';
import type { JsonTableFiles } from '@cfgforge/data';
import type { LangTextFinder } from './LangTextFinder.js';

export class ValueEnv {
  constructor(
    readonly fullSchema: CfgSchema,
    readonly cfgData: CfgData,
    readonly headRow: HeadRow,
    readonly nullableLangTextFinder: LangTextFinder | null,
    readonly jsonTableFiles: JsonTableFiles,
  ) {}
}
